### DISCUSSION TITLE
[Unknown Mod][Game] Enhancement to Windows related to Marketplace window and to Marketplace window

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3425] [Ideas]
For the wandering creatures Rally window, move the "Hero" button between the "YES" and "NO" buttons.
I think the text "No room in garrison" is not necessary and can be removed (the "Hero" button appears only when there is no room in the hero's army).

I also suggest to change the text for the "Marketplace" button to:

"Gold needed:
xxxxx"

Instead of:

"Not enough gold (xxxxx)"

![Pay to not fight fh2 01](https://user-images.githubusercontent.com/43583994/117549781-976d9680-b03c-11eb-95ed-58488b143f0c.png)

When the player accesses the Marketplace window because he does not have enough gold to perform an action (wandering creatures Rally window, Surrender window, Deamon Cave window (https://github.com/ihhub/fheroes2/discussions/3326), buy a Magic Book from a Shrine (https://github.com/ihhub/fheroes2/discussions/3424)), I suggest to clearly indicate the missing amount of gold.

The amount of gold is first indicated with a red text, and is continuously updated, changing to a green text when the required amount (or more) is reached.

Picture 1.
(1) A sentence is added between the title and the text (eventualy in small font) with an empty line before and after, with the amount of Gold in red color.
With simplified text compared to the one on the screen:
"Gold needed: xxxxx"
![Heroes 2 Screen 34](https://user-images.githubusercontent.com/43583994/118409683-06c23680-b68c-11eb-93f4-fc6a817f8057.png)

Picture 2.
(1) The amount of Gold is displayed in green color when the required amount (or more) is reached.
With simplified text compared to the one on the screen:
"Gold needed: xxxxx"
![Heroes 2 Screen 34b](https://user-images.githubusercontent.com/43583994/118409697-1477bc00-b68c-11eb-848a-471cc2a26785.png)

Picture 3.
The amount of gold in brackets is also shown in red (and green) and updated continuously for the gold total.
![Join marketplace red flèche 03](https://user-images.githubusercontent.com/43583994/118409742-3c671f80-b68c-11eb-9897-c65c936be839.png)

In the surrender window, when the player does not have enough gold, the text is also changed to:

"Gold needed: xxxxx"

Instead of:

"Not enough gold (xxxxx)"
![Trade gold info 001](https://user-images.githubusercontent.com/43583994/118409882-e3e45200-b68c-11eb-8d2f-e5c4acf22424.png)

Screens are for explanation only.

This is not an option.
This can be implemented directly in fheroes2.

**Comment by LeHerosInconnu** [comment]
In the Rally window, I think a "let them flee" button should be added in case the hero has Diplomacy and has an army with more than five times the strength of the wandering creatures (so when the hero doesn't want to recruit or fight the wandering creatures but they offer to join him).
The "Continue Movement" button could be used for this purpose.

![Hero Marketplace Flee buttons modifié 003](https://user-images.githubusercontent.com/43583994/180603967-bb16d7bb-022b-4dfa-b6df-fda3eb405cf6.png)

A right-click information window should also be added for the buttons in this window.

Screen is for explanation only.

**Reply by zenseii** [reply]
Hi @LeHerosInconnu ! A sort of related question, but are you always asked to pay gold or could upper tier creatures require rare resources?

**Reply by oleg-derevenetz** [reply]
@zenseii when wandering monsters join thanks to diplomacy, they only want gold.
